### PR TITLE
Refactorization : renaming some of the classes

### LINKED ISSUE #120: Improve constants name and function names

### PR DESCRIPTION
**Pull request by PonteIneptique**
I'll squash and merge but would be happy to get your review

Fixed #120

### CODE REVIEW

File: MyCapytain/common/metadata.py [hunk 1]
```diff
@@ -139,5 +140,5 @@ def getOr(subject, predicate, *args, **kwargs):
 
         """
         if (subject, predicate, None) in get_graph():
-            return Metadata(node=get_graph().objects(subject, predicate).__next__())
+            return Metadata(node=get_graph() .objects(subject, predicate).__next__())
```
Comment (sonofmun) [inline MyCapytain/common/metadata.py]: It looks like you added a space here. I don't think that should be there.

File: MyCapytain/resources/collections/cts.py [hunk 2]
```diff
@@ -185,30 +185,30 @@ def parse(resource, parent=None):
         :param resource: Element rerpresenting a work
         :param type: basestring, etree._Element
         :param parent: Parent of the object
-        :type parent: TextGroup
+        :type parent: XmlCtsTextgroupMetadata
```
Comment (sonofmun) [inline MyCapytain/resources/collections/cts.py]: Should Group be capitalized here? I.e., XmlCtsTextGroupMetadata? Maybe not worth the trouble. Or maybe not even desired.

File: CHANGES.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+### 2017-04-21 @ponteineptique
+
+Made a lot of renaming for clarity :
+
+| Old name | New name |
+|--------------|---------------|
+| MyCapytain.common.constants.Exportable | MyCapytain.common.base.Exportable | 
+| MyCapytain.common.constants.NS | MyCapytain.common.constants.XPath_Namespaces |
+| MyCapytain.common.constants.NAMESPACES | MyCapytain.common.constants.RDF_Namespaces |
```
Comment (sonofmun) [inline CHANGES.md]: Should RDF_Namepaces be all caps since it is a constance, i.e., RDF_NAMESPACES?

File: README.rst [hunk 4]
```diff
@@ -16,8 +16,8 @@
    :target: https://www.codacy.com/app/leponteineptique/MyCapytain
 
 
-MyCapytain is a python library which provides a large set of methods to interact with Text Services API  such as the \
-Canonical Text Services, the Distributed Text Services. It also provides a programming interface to exploit local \
+MyCapytain is a python library which provides a large set of methods to interact with CapitainsCtsText Services API  such as the \
+Canonical CapitainsCtsText Services, the Distributed CapitainsCtsText Services. It also provides a programming interface to exploit local \
```
Comment (sonofmun) [inline README.rst]: Did you really want to change Text here to CapitainCtsText? I would think that you actually want to leave these as Canonical Text Services"

File: doc/Dispatcher.py [hunk 5]
```diff
@@ -1,17 +1,17 @@
-from MyCapytain.resolvers.cts.local import CTSCapitainsLocalResolver
+from MyCapytain.resolvers.cts.local import CtsCapitainsLocalResolver
 from MyCapytain.resolvers.utils import CollectionDispatcher
-from MyCapytain.common.constants import Mimetypes, NS
-from MyCapytain.resources.collections.cts import TextInventory
-from MyCapytain.resources.prototypes.cts.inventory import TextInventoryCollection
+from MyCapytain.common.constants import Mimetypes
```
Comment (sonofmun) [inline doc/Dispatcher.py]: Should XPath_Namespaces (formerly NS) still be imported here?

File: tests/resources/collections/test_cts.py [hunk 6]
```diff
@@ -313,17 +313,16 @@ def test_Inventory_pickle(self):
...
             ))),
             1,
             "There should be one node for the child 1294 which is metadata"
         )
         self.assertEqual(*compareSTR(tixml, xml))
 
     def test_Inventory_metadata(self):
-        """ Tests PrototypeTextInventory parses without errors """
-        TI = TextInventory.parse(resource=self.getCapabilities)
-        print(TI["urn:cts:latinLit:phi1294.phi002.perseus-eng3"].get_link(constants.NAMESPACES.CTS.term("about")))
```
Comment (sonofmun) [inline tests/resources/collections/test_cts.py]: Sorry. I left that in there.

File: tests/resources/texts/api/test_cts.py [hunk 7]
```diff
@@ -676,8 +676,8 @@ def test_parentId(self):
             "ParentId should resolve"
         )
         self.assertIsInstance(
-            passage.parent, Passage,
-            "Parent Passage should be a passage"
+            passage.parent, CtsPassage,
+            "Parent CapitainsCtsPassage should be a passage"
```
Comment (sonofmun) [inline tests/resources/texts/api/test_cts.py]: Are CtsPassage and CapitainsCtsPassage the same thing? This makes it look like they are.

Comment (PonteIneptique) [inline tests/resources/texts/api/test_cts.py]: The second edit seems to be a mistake.

File: tests/testing_data/cts/getCapabilities.xml [hunk 8]
```diff
@@ -224,7 +224,7 @@
                     <ti:edition workUrn="urn:cts:latinLit:phi1276.phi001" urn="urn:cts:latinLit:phi1276.phi001.perseus-lat2">
                         <ti:label xml:lang="eng">Satires</ti:label>
                         <ti:description xml:lang="eng">
-            Juvenal and Persius: With An English Translation, G. G. Ramsay, 1918, William Heinemann; G. P. Putnam's Son, London
+            Juvenal and Persius: With An English XmlCtsTranslationMetadata, G. G. Ramsay, 1918, William Heinemann; G. P. Putnam's Son, London
```
Comment (sonofmun) [inline tests/testing_data/cts/getCapabilities.xml]: I am pretty sure you didn't want to change this. It does make for some funny metadata, however.

File: tests/testing_data/farsiLit/data/hafez/divan/hafez.divan.perseus-eng1.xml [hunk 9]
```diff
@@ -5657,7 +5657,7 @@
                         </l>
                         <l n="11">
                             <seg n="1">Hafiz ! if thou hear this royal counsel, ### </seg>
-                            <seg n="2">Passage to the highway of Tarikat thou canst make. *** </seg>
+                            <seg n="2">CapitainsCTSPassage to the highway of Tarikat thou canst make. *** </seg>
```
Comment (sonofmun) [inline tests/testing_data/farsiLit/data/hafez/divan/hafez.divan.perseus-eng1.xml]: This also shouldn't have been changed. These changes make me laugh, however.

File: tests/testing_data/farsiLit/data/hafez/divan/hafez.divan.perseus-eng1.xml [hunk 10]
```diff
@@ -9588,7 +9588,7 @@
                         </l>
                         <l n="7">
                             <seg n="1">By the Beloved's soul (I swear) that grief rendeth not your screen, ### </seg>
-                            <seg n="2">If, on the bounties of the Work-performer (God), reliance' make ye. *** </seg>
+                            <seg n="2">If, on the bounties of the XmlCtsWorkMetadata-performer (God), reliance' make ye. *** </seg>
```
Comment (sonofmun) [inline tests/testing_data/farsiLit/data/hafez/divan/hafez.divan.perseus-eng1.xml]: Change me back!!!!! God is not an XmlCtsWorkMetadata-performer!

File: tests/testing_data/farsiLit/data/hafez/divan/hafez.divan.perseus-eng1.xml [hunk 11]
```diff
@@ -17660,7 +17660,7 @@
                         </l>
                         <l n="8">
                             <seg n="1">Hafiz ! from kings, shelter in service they seek ; ### </seg>
-                            <seg n="2">Work not done, hope of gift what (is it that) thou holdest ? *** </seg>
+                            <seg n="2">XmlCtsWorkMetadata not done, hope of gift what (is it that) thou holdest ? *** </seg>
```
Comment (sonofmun) [inline tests/testing_data/farsiLit/data/hafez/divan/hafez.divan.perseus-eng1.xml]: Change back.

File: tests/testing_data/latinLit/data/phi0914/phi001/__cts__.xml [hunk 12]
```diff
@@ -3,8 +3,8 @@
       <ti:translation projid="latinLit:perseus-eng3" urn="urn:cts:latinLit:phi0914.phi001.perseus-eng3" workUrn="urn:cts:latinLit:phi0914.phi001" xml:lang="eng">
         <ti:label xml:lang="eng">The History of Rome</ti:label>
         <ti:description xml:lang="eng">Perseus:bib:oclc,2311635, Livy. History of Rome. English
-          Translation by. Rev. Canon Roberts. New York, New York. E. P. Dutton and Co. 1912. 1.
-          Livy. History of Rome. English Translation. Rev. Canon Roberts. New York, New York. E. P.
+          XmlCtsTranslationMetadata by. Rev. Canon Roberts. New York, New York. E. P. Dutton and Co. 1912. 1.
```
Comment (sonofmun) [inline tests/testing_data/latinLit/data/phi0914/phi001/__cts__.xml]: Change these two lines back.

File: tests/testing_data/latinLit/data/phi0959/phi006/__cts__.xml [hunk 13]
```diff
@@ -10,14 +10,14 @@
   </ti:edition>
   <ti:translation urn="urn:cts:latinLit:phi0959.phi006.perseus-eng3"
     workUrn="urn:cts:latinLit:phi0959.phi006" xml:lang="eng">
-    <ti:label xml:lang="eng">Metamorphoses (English Translation by Brooks More)</ti:label>
+    <ti:label xml:lang="eng">Metamorphoses (English XmlCtsTranslationMetadata by Brooks More)</ti:label>
```
Comment (sonofmun) [inline tests/testing_data/latinLit/data/phi0959/phi006/__cts__.xml]: Change back.

File: tests/testing_data/latinLit/data/phi0959/phi006/__cts__.xml [hunk 14]
```diff
@@ -10,14 +10,14 @@
...
     <ti:description xml:lang="eng">Perseus:bib:oclc,24965574, Ovid. Metamorphoses. Brookes More.
       Boston. Cornhill Publishing Co. 1922.</ti:description>
     <ti:memberof collection="Perseus:collection:Greco-Roman"/>
   </ti:translation>
   <ti:translation urn="urn:cts:latinLit:phi0959.phi006.perseus-eng4"
     workUrn="urn:cts:latinLit:phi0959.phi006" xml:lang="eng">
-    <ti:label xml:lang="eng">Metamorphoses (English Translation by Golding Arthur)</ti:label>
+    <ti:label xml:lang="eng">Metamorphoses (English XmlCtsTranslationMetadata by Golding Arthur)</ti:label>
```
Comment (sonofmun) [inline tests/testing_data/latinLit/data/phi0959/phi006/__cts__.xml]: Change back.

File: tests/testing_data/latinLit/data/phi1276/phi001/__cts__.xml [hunk 15]
```diff
@@ -3,7 +3,7 @@
     <ti:edition workUrn="urn:cts:latinLit:phi1276.phi001" urn="urn:cts:latinLit:phi1276.phi001.perseus-lat2">
         <ti:label xml:lang="eng">Satires</ti:label>
         <ti:description xml:lang="eng">
-            Juvenal and Persius: With An English Translation, G. G. Ramsay, 1918, William Heinemann; G. P. Putnam's Son, London
+            Juvenal and Persius: With An English XmlCtsTranslationMetadata, G. G. Ramsay, 1918, William Heinemann; G. P. Putnam's Son, London
```
Comment (sonofmun) [inline tests/testing_data/latinLit/data/phi1276/phi001/__cts__.xml]: Change back.

File: tests/testing_data/latinLit2/data/phi0959/phi001/phi0959.phi001.perseus-eng2.xml [hunk 16]
```diff
@@ -393,7 +393,7 @@ schematypens="http://purl.oclc.org/dsdl/schematron"?><TEI xmlns="http://www.tei-
 					<l n="33">Think what soft things to move her soul I said,</l>
 					<l n="34">And let them in a lover's favour plead.</l>
 					<l n="35">But ah! the tender things that made her kind,</l>
-					<l n="36">Work no such wonders on thy cruel mind.</l>
+					<l n="36">XmlCtsWorkMetadata no such wonders on thy cruel mind.</l>
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/phi0959/phi001/phi0959.phi001.perseus-eng2.xml]: Change back.

File: tests/testing_data/latinLit2/data/phi0959/phi006/__cts__.xml [hunk 17]
```diff
@@ -10,14 +10,14 @@
   </ti:edition>
   <ti:translation urn="urn:cts:latinLit:phi0959.phi006.perseus-eng3"
     workUrn="urn:cts:latinLit:phi0959.phi006" xml:lang="eng">
-    <ti:label xml:lang="eng">Metamorphoses (English Translation by Brooks More)</ti:label>
+    <ti:label xml:lang="eng">Metamorphoses (English XmlCtsTranslationMetadata by Brooks More)</ti:label>
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/phi0959/phi006/__cts__.xml]: Change back.

File: tests/testing_data/latinLit2/data/phi0959/phi006/__cts__.xml [hunk 18]
```diff
@@ -10,14 +10,14 @@
...
     <ti:description xml:lang="eng">Perseus:bib:oclc,24965574, Ovid. Metamorphoses. Brookes More.
       Boston. Cornhill Publishing Co. 1922.</ti:description>
     <ti:memberof collection="Perseus:collection:Greco-Roman"/>
   </ti:translation>
   <ti:translation urn="urn:cts:latinLit:phi0959.phi006.perseus-eng4"
     workUrn="urn:cts:latinLit:phi0959.phi006" xml:lang="eng">
-    <ti:label xml:lang="eng">Metamorphoses (English Translation by Golding Arthur)</ti:label>
+    <ti:label xml:lang="eng">Metamorphoses (English XmlCtsTranslationMetadata by Golding Arthur)</ti:label>
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/phi0959/phi006/__cts__.xml]: Change back.

File: tests/testing_data/latinLit2/data/phi0959/phi006/phi0959.phi006.perseus-eng3.xml [hunk 19]
```diff
@@ -84,7 +84,7 @@
 
 <revisionDesc>
 <change when="2014-11-06" who="Thibault Clerice">Moving to Epidoc with new URN</change>
-<change when="2013-09-13" who="balmas01">moving Greg's work-in-progress on CTS-izing greco-roman collection texts to a Work directory</change>
+<change when="2013-09-13" who="balmas01">moving Greg's work-in-progress on CTS-izing greco-roman collection texts to a XmlCtsWorkMetadata directory</change>
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/phi0959/phi006/phi0959.phi006.perseus-eng3.xml]: Change back.

File: tests/testing_data/latinLit2/data/phi0959/phi006/phi0959.phi006.perseus-eng4.xml [hunk 20]
```diff
@@ -83,7 +83,7 @@
 
 <revisionDesc>
   <change when="2014-11-06" who="Thibault Clerice">Moving to Epidoc with new URN</change>
-  <change when="2013-09-13" who="balmas01">moving Greg's work-in-progress on CTS-izing greco-roman collection texts to a Work directory</change>
+  <change when="2013-09-13" who="balmas01">moving Greg's work-in-progress on CTS-izing greco-roman collection texts to a XmlCtsWorkMetadata directory</change>
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/phi0959/phi006/phi0959.phi006.perseus-eng4.xml]: Change back.

File: tests/testing_data/latinLit2/data/phi0959/phi006/phi0959.phi006.perseus-lat2.xml [hunk 21]
```diff
@@ -79,7 +79,7 @@ schematypens="http://purl.oclc.org/dsdl/schematron"?>
       </profileDesc>
 
       <revisionDesc>
-         <change who="#balmas01" when="2013-09-13T11:17:41">moving Greg's work-in-progress on CTS-izing greco-roman collection texts to a Work directory</change>
+         <change who="#balmas01" when="2013-09-13T11:17:41">moving Greg's work-in-progress on CTS-izing greco-roman collection texts to a XmlCtsWorkMetadata directory</change>
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/phi0959/phi006/phi0959.phi006.perseus-lat2.xml]: Change back.

File: tests/testing_data/latinLit2/data/phi1276/phi001/__cts__.xml [hunk 22]
```diff
@@ -3,7 +3,7 @@
     <ti:edition workUrn="urn:cts:latinLit:phi1276.phi001" urn="urn:cts:latinLit:phi1276.phi001.perseus-lat2">
         <ti:label xml:lang="eng">Satires</ti:label>
         <ti:description xml:lang="eng">
-            Juvenal and Persius: With An English Translation, G. G. Ramsay, 1918, William Heinemann; G. P. Putnam's Son, London
+            Juvenal and Persius: With An English XmlCtsTranslationMetadata, G. G. Ramsay, 1918, William Heinemann; G. P. Putnam's Son, London
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/phi1276/phi001/__cts__.xml]: Change back.

File: tests/testing_data/latinLit2/data/phi1276/phi001/phi1276.phi001.perseus-lat2.xml [hunk 23]
```diff
@@ -26,7 +26,7 @@
 				        <biblStruct>
 					          <monogr>
 						            <author>Juvenal</author>
-						            <title>Juvenal and Persius: With An English Translation</title>
+						            <title>Juvenal and Persius: With An English XmlCtsTranslationMetadata</title>
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/phi1276/phi001/phi1276.phi001.perseus-lat2.xml]: Change back.

File: tests/testing_data/latinLit2/data/tlg0003/tlg001/tlg0003.tlg001.opp-ger1.xml [hunk 24]
```diff
@@ -105,7 +105,7 @@
             doch immer groß genug, um für unsere Gegenwart <lb/> reiche Belehrung und Zurechtweisung
             zu bieten, und das <lb/> Geschichtswerk des Thukydides ist das Buch, welches solche am
             <lb/> reichlichsten enthält.</p>
-               <p>Dieser Uebersetzung liegt vorzugsweise der Text und <lb/> die Texterklärung der
+               <p>Dieser Uebersetzung liegt vorzugsweise der CapitainsCTSText und <lb/> die Texterklärung der
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/tlg0003/tlg001/tlg0003.tlg001.opp-ger1.xml]: Change back.

File: tests/testing_data/latinLit2/data/tlg0003/tlg001/tlg0003.tlg001.opp-ger1.xml [hunk 25]
```diff
@@ -11247,7 +11247,7 @@
               nicht stattgefunden hatten. ' i <note type="footnote"> 52) Fest der tphesischen
                 Artemis. Strabo S. S4S. </note>
                      <note type="footnote"> 53) I Hymne aus Apollo V. He.
-                165. Der bei Thuk. gegebene Text ist <lb/> von den gewöhnlichen etwas abweichend.
+                165. Der bei Thuk. gegebene CapitainsCTSText ist <lb/> von den gewöhnlichen etwas abweichend.
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/tlg0003/tlg001/tlg0003.tlg001.opp-ger1.xml]: Change back.

File: tests/testing_data/latinLit2/data/tlg0003/tlg001/tlg0003.tlg001.opp-ger2.xml [hunk 26]
```diff
@@ -9459,7 +9459,7 @@
             natürlichen Neigungen der Menschen, die schon gewohnt sind, <lb/>
             auch bei wirklicher Gültigkeit der Gesetze doch dagegen zu <lb/>
             sündigen, gänzlich besiegt wurden, zeigte sich's frei, daß der <note type="footnote"> 1 Kap. 84 gilt schon lange als »neckt. Ich verzichte auf einen neuen Versuch, <lb/>
-            eS in lesbares Deutsch zu übertragen, und gebe dafür im Text die Heil­ <lb/>
+            eS in lesbares Deutsch zu übertragen, und gebe dafür im CapitainsCTSText die Heil­ <lb/>
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/tlg0003/tlg001/tlg0003.tlg001.opp-ger2.xml]: Change back.

File: tests/testing_data/latinLit2/data/tlg0003/tlg001/tlg0003.tlg001.opp-ger3.xml [hunk 27]
```diff
@@ -11496,7 +11496,7 @@
             Wiederherstellung <lb/> 8, 86 -, inKerkyra 70; in Argos <lb/> u. Mantinea 5,47; in Chios
             8,14.</p>
           <p>Rathaus (Buleuterion) in Athen 8,00.</p>
-          <p>Reden, ihre Wiedergabe im Text <lb/> 1,22; des Alkibiades 6,16 ff., soff; <lb/>
+          <p>Reden, ihre Wiedergabe im CapitainsCTSText <lb/> 1,22; des Alkibiades 6,16 ff., soff; <lb/>
```
Comment (sonofmun) [inline tests/testing_data/latinLit2/data/tlg0003/tlg001/tlg0003.tlg001.opp-ger3.xml]: Change back.

File: CHANGES.md [hunk 28]
```diff
@@ -0,0 +1,33 @@
+### 2017-04-21 @ponteineptique
+
+Made a lot of renaming for clarity :
+
+| Old name | New name |
+|--------------|---------------|
+| MyCapytain.common.constants.Exportable | MyCapytain.common.base.Exportable | 
+| MyCapytain.common.constants.NS | MyCapytain.common.constants.XPath_Namespaces |
```
Comment (sonofmun) [inline CHANGES.md]: Also here, perhaps it should be XPATH_NAMESPACES?

File: MyCapytain/common/metadata.py (hunk 1 above)
Comment (PonteIneptique) [inline MyCapytain/common/metadata.py]: Done

File: MyCapytain/resources/collections/cts.py (hunk 2 above)
Comment (PonteIneptique) [inline MyCapytain/resources/collections/cts.py]: I think capitalized textgroup is a hassle and a mistake I did :)

File: doc/Dispatcher.py (hunk 5 above)
Comment (PonteIneptique) [inline doc/Dispatcher.py]: It was a mistakenly imported module

**Comment by PonteIneptique** [comment]
@balmas You are free to take a look, albeit @sonofmun did the heavy reviewing :)

**Comment by balmas** [comment]
ok, will try to get to it...

**Pull request merged**